### DISCUSSION TITLE
Static library libglad.a has to be generated manually

### DISCUSSION THREAD

**Discussion by knitterJ** [Dav1dde/glad#378] [Q&A] [answered]
Can I receive explanation why do I have to generate static library file manually and why it wasn't possible to add lib folder with the file like libglad.a for the generation process?

**Comment by Dav1dde** [comment, marked as answer]
Two main reasons:

Glad is not supposed to be built/ran everytime you build your application. The intended way is to include the generated code in your project and only update it (or re-generate it), when you actually need to do that (one of the reasons I dont really like the CMake integration). This is also why there is a web generator, so you don't even have to have Python installed.

The other reason is, it simply would not be possible. Compiling code is different on basically every system (that's why autoconfigure, cmake and all these other systems exist). It would be impossible for me to actually compile the code in a way it works *everywhere*.

**Comment by RohitRam9701** [comment]
To generate a static library `libglad.a` manually, you need to follow these steps:

1. Go to the GLAD website (https://glad.dav1d.de/).
2. Configure the generator options:
   - Select the language as "C/C++."
   - Select the API as "OpenGL."
   - Choose the appropriate version of OpenGL for your project.
   - Check the "Generate a loader" option for GLAD.
   - Set the "Profile" to "Core" or "Compatibility" depending on your needs.
   - Make sure to leave the "Extensions" option as "All."
3. Click the "Generate" button to download the zip file containing the generated GLAD source and header files.
4. Extract the contents of the downloaded zip file to a location on your computer.

Now, follow these steps to create the static library manually:

1. Open a terminal or command prompt.
2. Navigate to the directory where you extracted the GLAD files using the `cd` command. For example:
   ```
   cd /path/to/glad/files
   ```
3. Compile the `glad.c` source file into an object file (`.o`). Use the appropriate compiler and flags for your platform. For MinGW on Windows, you can use the following command:
   ```
   gcc -c glad.c -o glad.o
   ```
   This will generate the `glad.o` object file.

4. Create the static library `libglad.a` from the object file:
   ```
   ar rcs libglad.a glad.o
   ```
   This will generate the `libglad.a` static library file.

Now, you have successfully generated the `libglad.a` static library, which you can use in your projects. Remember to link this library along with other required libraries when compiling your OpenGL programs.

Additionally, make sure you include the appropriate GLAD header in your source code. For example, in your C/C++ source files, include the following line at the top:

```c
#include <glad/glad.h>
```

With these steps completed, you should be able to compile and link your OpenGL programs with the manually generated `libglad.a` static library.
It worked for me very well.
